src/libs/antares/study/scenario-builder/rules.cpp

```diff
@@ -201,6 +205,33 @@ namespace ScenarioBuilder
 		}
 	}
 
+	void Rules::sendWarningsForDisabledClusters()
+	{
+		map < string, vector<uint> >::iterator it = disabledClustersOnRuleActive.begin();
+		for (; it != disabledClustersOnRuleActive.end(); it++)
```

Comment:
On one hand, your proposal leads to shorter code, indeed.
On the other hand, I'm not a fan of **auto** keyword, because it hides the real type of **disabledClustersOnRuleActive**.
In this case, keeping one line doesn't seem that ugly.
What do you think ?

Reply:
Declaring 'it' before the for loop maintains its existence after the loop, so this solution is not entirely satisfying.

I suggest instead

`for (map < string, vector<uint> >::iterator it = disabledClustersOnRuleActive.begin(); it != disabledClustersOnRuleActive.end(); it++)`

or you can add accolades around the declaration of 'it' and the loop.